src/AutofacSerilogIntegration/ContextualLoggingModule.cs

```diff
@@ -48,8 +50,17 @@ protected override void Load(ContainerBuilder builder)
 
                 return logger;
             })
-            .As<ILogger>()
-            .ExternallyOwned();
+            .As<ILogger>();
+
+            if (_dispose)
+            {
+                registration.SingleInstance()
```

Comment:
I've only tested it locally on a web project, where it worked fine. It is true though that I may be too focused on my own use case.

Based on [Lifecycle of Loggers](https://github.com/serilog/serilog/wiki/Lifecycle-of-Loggers) I have the understanding that only the root logger ever needs disposing of (due to IDisposable sinks). Thus, if the lifecycle of the root logger starts with the container and is disposed by the container when the root container itself is disposed (application end), I don't see why this wouldn't have the desired effect?

The way I've tested this is by adding a breakpoint in the web application application_end event and observing the Log.Logger property. With dispose set to false, the logger was still an instance of `Serilog.Core.Logger`, with dispose set to true it was instead an instance of `Serilog.Core.Pipeline.SilentLogger`.

I see a potential issue with the user passing in a custom ILogger into the RegisterLogger extension method (i. e. one which the container is not supposed to be disposing) together with `dispose: true` - which could result in the container disposing the root logger unexpectedly (from the consuming application's point of view). Maybe add an additional guard against that?